File: src/main/java/io/tripled/social/client/application/DefaultReadMessagesUseCase.java
package io.tripled.social.client.application;

import io.tripled.social.client.domain.SocialNetwork;
import io.tripled.social.client.domain.SocialNetworkRepository;
import io.tripled.social.client.domain.UserName;

public class DefaultReadMessagesUseCase implements ReadMessagesUseCase {

  private SocialNetworkRepository socialNetworkRepository;

  public DefaultReadMessagesUseCase(SocialNetworkRepository socialNetworkRepository) {
    this.socialNetworkRepository = socialNetworkRepository;
  }

  @Override
  public String readMessage(String userName) {
    SocialNetwork socialNetwork = socialNetworkRepository.get();

    return socialNetwork.read(new UserName(userName));
  }
}

File: src/main/kotlin/io/tripled/social/client/application/DefaultReadMessagesUseCase.kt
package io.tripled.social.client.application

import io.tripled.social.client.domain.SocialNetwork
import io.tripled.social.client.domain.SocialNetworkRepository
import io.tripled.social.client.domain.UserName

class DefaultReadMessagesUseCase(socialNetworkRepository: SocialNetworkRepository) : ReadMessagesUseCase {
    private val socialNetworkRepository: SocialNetworkRepository

    init {
        this.socialNetworkRepository = socialNetworkRepository
    }

    override fun readMessage(userName: String): String {
        val socialNetwork: SocialNetwork = socialNetworkRepository.get()

        return socialNetwork.read(UserName(userName))
    }
}
